Backbase/variants
# Allow setting iOS Bundle id for Variant

### What does this PR do
Allows user to set the BundleID for a specific variant.

### Extra changes
- Moved signing validation from Configuration to the iOSVariant object itself.
- Fixed typo with a misleading instruction present in the variants config file template
- Clean up BB Team ID from test data

### How can it be tested
> AC 1:
> GIVEN I want to fully customize my app variant BundleID
> WHEN I add the key `bundle_id` to the variant config
> AND I run `switch` command to the specific variant
> THEN I should see my app with the exact BundleID as configured


> AC 2:
> GIVEN I added both `bundle_id` and `id_suffix` to my variant config
> WHEN I run `switch` command to the specific variant
> THEN I should get an error as both configs are not compatible at same time

### Task
Resolves #159 

### Checklist:
